File: frontend/src/api/modules/dataQuality.ts
/**
 * 数据质量 API
 */
import request from '@/utils/request'

export interface DataQualityPointInfo {
  point_id: number
  point_code: string
  point_name: string
  device_type: string | null
  quality: number
  quality_text: string
  status: string | null
  updated_at: string | null
}

export interface DataQualityStatus {
  total: number
  normal_count: number
  uncertain_count: number
  unreliable_count: number
  unreliable_points: DataQualityPointInfo[]
}

export function getDataQualityStatus() {
  return request.get<DataQualityStatus>('/v1/data-quality/status')
}

export function getDataQualityPoints(params?: { quality?: number }) {
  return request.get<DataQualityPointInfo[]>('/v1/data-quality/points', { params })
}

File: backend/app/schemas/data_quality.py
"""数据质量相关 Schema"""

from pydantic import BaseModel, ConfigDict
from typing import List, Optional
from datetime import datetime


class DataQualityPointInfo(BaseModel):
    point_id: int
    point_code: str
    point_name: str
    device_type: Optional[str] = None
    quality: int
    quality_text: str
    status: Optional[str] = None
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)


class DataQualityStatus(BaseModel):
    total: int
    normal_count: int
    uncertain_count: int
    unreliable_count: int
    unreliable_points: List[DataQualityPointInfo] = []
